util/config/appdir.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2018 Shift Devices AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"os"
+	"path/filepath"
+	"runtime"
+)
+
+// AppDir returns the absolute path to the default BitBox desktop app directory
+// in the user standard config location.
+func AppDir() string {
+	var appFolder string
+	switch runtime.GOOS {
+	case "darwin":
+		// Usually /Users/$USER/Library/Application Support.
+		appFolder = os.Getenv("HOME") + "/Library/Application Support"
+	case "windows":
+		appFolder = os.Getenv("APPDATA")
+	case "linux":
+		// Previously, we always used $HOME/.config/bitbox to store paired channel data.
```

Comment:
I guess it's possible but what do you do in case of an error? Panic, keep things as is, display an error? Unclear.

It's true that a user would loose their desktop app data, but at least they won't have to reset their device in case 2FA was enabled. I guess it's either one or the other. Which one would you choose?

Reply:
The issue with "moving" a file is what happens if [os.Rename](https://golang.org/pkg/os/#Rename) returns an error. Did it actually get to moving the file or left it in the old place? Also depends on the OS. Although it is probably a single syscall on all supported platforms, the implementation of that syscall is different, e.g. windows vs unix. 

You could say, let's copy it first then remove the old one. But this is not atomic. So you might end up with 2 copies... or worse: one of them is corrupted.

On top of that, add multiple goroutines calling `config.AppDir` at the same time and I suspect this might become a big nightmare. :(

Reply:
BTW, that's why I wanted to have a separate pull request for this. Had a feeling it wouldn't be that easy.

Reply:
I also wrote https://github.com/x1ddos/bitbox-wallet-app/commit/f4eee89287cd18f3dcb4fca704cd65e98e7a7d53 which includes tests for `config.AppDir` but will send it after this PR is merged, or feel free to cherry-pick whenever is best.